From 16b1a65206e85f718d920cfec7dd0fd1c0995d19 Mon Sep 17 00:00:00 2001
From: Jonas Karlsson <jonas.karlsson@qt.io>
Date: Thu, 5 May 2022 07:13:16 +0200
Subject: [PATCH] Fix build error with arm64

---
 .../foundation/include/unix/neon/PsUnixNeonInlineAoS.h     | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/3rdparty/PhysX/source/foundation/include/unix/neon/PsUnixNeonInlineAoS.h b/src/3rdparty/PhysX/source/foundation/include/unix/neon/PsUnixNeonInlineAoS.h
index 81b7a99..8a1059d 100644
--- a/src/3rdparty/PhysX/source/foundation/include/unix/neon/PsUnixNeonInlineAoS.h
+++ b/src/3rdparty/PhysX/source/foundation/include/unix/neon/PsUnixNeonInlineAoS.h
@@ -3567,7 +3567,6 @@ PX_FORCE_INLINE VecU32V V4U32SplatElement(VecU32V a)
 template <int index>
 PX_FORCE_INLINE Vec4V V4SplatElement(Vec4V a)
 {
-#if PX_UWP
 	if(index == 0)
 	{
 		return vdupq_lane_f32(vget_low_f32(a), 0);
@@ -3576,12 +3575,6 @@ PX_FORCE_INLINE Vec4V V4SplatElement(Vec4V a)
 	{
 		return vdupq_lane_f32(vget_low_f32(a), 1);
 	}
-#else
-	if(index < 2)
-	{
-		return vdupq_lane_f32(vget_low_f32(a), index);
-	}
-#endif
 	else if(index == 2)
 	{
 		return vdupq_lane_f32(vget_high_f32(a), 0);
-- 
2.34.0

