--- .upstream/moc.cpp	2023-08-11 10:37:23.940392525 +0200
+++ moc.cpp	2023-08-21 11:21:53.873272207 +0200
@@ -12,14 +12,15 @@
 #include <QtCore/qdir.h>
 #include <QtCore/qjsondocument.h>
 
-// for normalizeTypeInternal
-#include <private/qmetaobject_moc_p.h>
-#include <private/qduplicatetracker_p.h>
+// -- QScxml
+#include <QtCore/qjsonobject.h>
+// -- QScxml
 
 QT_BEGIN_NAMESPACE
 
 using namespace Qt::StringLiterals;
 
+#if 0 // -- QtScxml
 // only moc needs this function
 static QByteArray normalizeType(const QByteArray &ba)
 {
@@ -1953,6 +1954,7 @@
         }
     }
 }
+#endif // -- QScxml
 
 QJsonObject ClassDef::toJson() const
 {
