--- .upstream/moc.cpp	2024-02-01 11:08:00.055494626 +0100
+++ moc.cpp	2024-02-05 14:06:04.258179059 +0100
@@ -12,14 +12,15 @@
 #include <QtCore/qdir.h>
 #include <QtCore/qjsondocument.h>
 
-// for normalizeTypeInternal
-#include <private/qmetaobject_moc_p.h>
-#include <private/qduplicatetracker_p.h>
+// -- QtScxml
+#include <QtCore/qjsonobject.h>
+// -- QtScxml
 
 QT_BEGIN_NAMESPACE
 
 using namespace Qt::StringLiterals;
 
+#if 0 // -- QtScxml
 // only moc needs this function
 static QByteArray normalizeType(const QByteArray &ba)
 {
@@ -1997,6 +1998,7 @@
         }
     }
 }
+#endif // -- QtScxml
 
 QJsonObject ClassDef::toJson() const
 {
