--- .upstream/moc.cpp	2024-08-05 12:14:28.111250492 +0200
+++ moc.cpp	2024-08-05 16:00:04.711392552 +0200
@@ -12,14 +12,15 @@
 #include <QtCore/qdir.h>
 #include <QtCore/qjsondocument.h>
 
-// for normalizeTypeInternal
-#include <private/qmetaobject_moc_p.h>
-#include <private/qduplicatetracker_p.h>
+// -- QtScxml
+#include <QtCore/qjsonobject.h>
+// -- QtScxml
 
 QT_BEGIN_NAMESPACE
 
 using namespace Qt::StringLiterals;
 
+#if 0 // -- QtScxml
 // only moc needs this function
 static QByteArray normalizeType(const QByteArray &ba)
 {
@@ -1999,6 +2000,7 @@
         }
     }
 }
+#endif // -- QtScxml
 
 QJsonObject ClassDef::toJson() const
 {
